Added announcements module and configuration

Hardik-hi opened the pull request:
**Changes**

1. Added configuration schema for announcements module
2. Added announcements module
3. Added ViewAnnouncements page (for viewing announcements)
4. Added EditAnnouncements and CreateAnnouncement components


cc: @coolbung

Hardik-hi committed:
Added announcement config

Hardik-hi committed:
Added announcements module

Hardik-hi committed:
Added announcement service

Hardik-hi committed:
Added View Announcements component

Hardik-hi committed:
Added formatting

Hardik-hi committed:
Added EditAnnouncement, CreateAnnouncement

Hardik-hi committed:
Integrated announcement config with admin console

coolbung commented on modules/announcements/src/App.tsx:
```diff
@@ -0,0 +1,75 @@
+function App() {
+  const routes: RouteProps[] = [
+    {
+      path: "/",
```
The URLs that will be seen in the browser should be
/announcements
/announcements/create
/announcements/edit/:id

arajput replied:
this module is not yet used in console app. the these routes are only used when app runs as standalone.
the urls are 
https://sandbox.shikshaplatform.io/modules/announcements/...

coolbung replied:
Got it. Do update the routes anyway @Hardik-hi and also ensure appropriate routes are used when this module is imported into the parent admin ui app.

Hardik-hi replied:
Sure @coolbung

coolbung commented on modules/announcements/src/index.css:
```diff
@@ -0,0 +1,13 @@
+body {
```
Is the CSS necessary, does it not import the CSS from the parent app ? 

cc @arajput

arajput replied:
this is default file by CRA. it is used only when module runs as stand alone.

arajput commented on modules/announcements/src/pages/CreateAnnouncement.tsx:
```diff
@@ -0,0 +1,114 @@
+    <Box marginX={4}>
+      <Flex direction={"row"}>
+        <Heading as="h4" size="md">
+          Create Announcement
```
use translations for the text

Hardik-hi replied:
On it @arajput

Hardik-hi committed:
Route changes

Hardik-hi committed:
UI Changes, abtracted AnnouncementForm

Hardik-hi committed:
Added translations

Hardik-hi commented:
I am currently facing issues while adding the announcements module to admin console

I have added the Announcements module to the admin-console (inside Admin UI) but when using the useLocation() hook inside the component I am getting the following error:


1. I have checked the component tree and Router and BrowserRouter both are parents of the announcements main component. 

2. I have also added the required configuration similar to the one for config-ui inside the following files:

modules/announcements/src/App.tsx
apps/admin-console/src/App.jsx
apps/admin-console/public/modules.json
apps/admin-console/modulefederation.config.js

3. Announcements is working fine when running as a standalone module but not when included inside the admin console

![image](https://user-images.githubusercontent.com/55993168/181046973-0ffcdca8-e591-4309-b74c-14e17a4271dd.png)

CC: @coolbung  @arajput

Hardik-hi committed:
fix: changes wrt api

Hardik-hi committed:
feat: added toasts

Hardik-hi committed:
fix: integration in admin console

Hardik-hi commented:
@coolbung @arajput Have fixed integration changes, please review

Hardik-hi committed:
type changes

Hardik-hi committed:
fix: admin console merge error
